docs/src/reference/asciidoc/core/kerberos.adoc

```diff
@@ -344,16 +485,39 @@ in the Hadoop configurations, then the credential provider will assume that 'sim
 will not obtain delegation tokens.
 
 Secondly, all general connection settings for {eh} (like `es.nodes`, `es.ssl.enabled`, etc...) must be specified either
-in the local Hadoop configuration files, in the local Spark configuration file, or from the command line. If these
-settings are not available here, then the credential provider will not be able to contact {es} in order to obtain the
-delegation tokens that it requires.
+<<kerberos-yarn,in the local Hadoop configuration files>>, in the local Spark configuration file, or from the command
+line. If these settings are not available here, then the credential provider will not be able to contact {es} in order
+to obtain the delegation tokens that it requires.
+
+[source,bash]
+----
+$> bin/spark-submit \
+    --class org.myproject.MyClass \
+    --master yarn \
+    --deploy-mode cluster \
+    --jars path/to/elasticsearch-hadoop.jar \
+    --conf 'spark.es.nodes=es-node-1,es-node-2,es-node-3' <1>
+    --conf 'spark.es.ssl.enabled=true'
+    --conf 'spark.es.net.spnego.auth.elasticsearch.principal=HTTP/_HOST@REALM' <2>
+    path/to/jar.jar
+----
+<1> An example of some connection settings specified at submission time
+<2> Be sure to include the {es} service principal.
+
+NOTE: Specifying this many configurations in the spark-submit command line is a pretty sure fire way to miss important
+settings. Thus, it is advised to set them in the <<kerberos-yarn,cluster wide Hadoop config>>.
 
 [[kerberos-spark-credprovider-streaming]]
 ===== Renewing credentials for streaming jobs
 
 NOTE: In the event that you are running a streaming job, it is best to use the `cluster` deploy mode to allow YARN to
 manage running the driver code for the streaming application.
 
+The delegation tokens from {es} that are retrieved are capped at a maximum lifetime of 7 days. After that time passes,
+the tokens will no longer successfully authenticate connections. Since streaming jobs are expected to run continuously
+without stopping, you should configure Spark so that the credential provider can obtain new tokens before the original
```

Comment:
>> you should configure Spark so that the credential provider can obtain new tokens

I assume this is reference to the YARN configuration but the wording kinda sounds like the Java code (`org.apache.spark.deploy.yarn.security.ServiceCredentialProvider`).

Maybe just link to YARN config again ?

Reply:
Yeah, the wording here is similar but just slightly different than the YARN setup. For YARN, it can _renew_ and _cancel_ tokens, but it cannot _obtain_ new tokens. The Spark api has the ability to _obtain_ new tokens and then transmit them to each worker node via a shared directory on HDFS. The thing here is that Spark can only obtain those new tokens if the user gives it their Kerberos principal and keytab when they launch the job, since Spark will need to relogin to Kerberos to get new tokens.

I'll add a blurb explaining the difference between YARN and Spark in this section.